root/static/scripts/tests/Control/URLCleanup.js

```diff
@@ -1189,6 +1189,67 @@ limited_link_type_combinations: [
        input_relationship_type: 'otherdatabases',
        only_valid_entity_types: [],
   },
+  // Bugs
+  {
+                     input_url: 'https://music.bugs.co.kr/album/20488834?wl_ref=M_contents_01_07',
+             input_entity_type: 'release',
+    expected_relationship_type: ['downloadpurchase', 'streamingpaid'],
+limited_link_type_combinations: [
+                                  ['downloadpurchase', 'streamingpaid'],
+                                  'streamingpaid',
+                                ],
+            expected_clean_url: 'https://music.bugs.co.kr/album/20488834',
+  },
+  {
+                     input_url: 'https://m.bugs.co.kr/album/20488834?wl_ref=M_contents_01_07',
+             input_entity_type: 'release',
+    expected_relationship_type: ['downloadpurchase', 'streamingpaid'],
+limited_link_type_combinations: [
+                                  ['downloadpurchase', 'streamingpaid'],
+                                  'streamingpaid',
+                                ],
+            expected_clean_url: 'https://music.bugs.co.kr/album/20488834',
+  },
+  {
+
+                     input_url: 'https://music.bugs.co.kr/artist/80276288?wl_ref=M_Search_01_01',
+             input_entity_type: 'artist',
+    expected_relationship_type: ['downloadpurchase', 'streamingpaid'],
+limited_link_type_combinations: [
+                                  ['downloadpurchase', 'streamingpaid'],
+                                  'streamingpaid',
+                                ],
+            expected_clean_url: 'https://music.bugs.co.kr/artist/80276288',
+  },
+  {
+
+                     input_url: 'https://m.bugs.co.kr/artist/80276288',
+             input_entity_type: 'artist',
+    expected_relationship_type: ['downloadpurchase', 'streamingpaid'],
+limited_link_type_combinations: [
+                                  ['downloadpurchase', 'streamingpaid'],
+                                  'streamingpaid',
+                                ],
+            expected_clean_url: 'https://music.bugs.co.kr/artist/80276288',
+  },
+  {
+
+                     input_url: 'https://music.bugs.co.kr/search/integrated?q=dreamcatcher',
+             input_entity_type: 'artist',
+    expected_relationship_type: undefined,
+  },
```

Comment:
This test is good and can be improved further as follows:
- Checking that it isn’t valid for any entity type (using empty array with `only_valid_entity_types`)
- Checking that the URL is left untouched (using the input URL with `expected_clean_url`)
- Checking that the error message is specific (using an extract of it with `expected_error`)

```suggestion
  {
                     input_url: 'https://music.bugs.co.kr/search/integrated?q=dreamcatcher',
             input_entity_type: 'artist',
    expected_relationship_type: undefined,
       only_valid_entity_types: [],
            expected_clean_url: 'https://music.bugs.co.kr/search/integrated?q=dreamcatcher',
                expected_error: {
                                  error: 'a link to a search result',
                                  target: 'url',
                                },
  },
```

Reply:
I tried with `only_valid_entity_types` earlier, but my tests failed, probably because I'd also need to pass `input_relationship_type`. Should I just pass `streamingpaid` in that case?

And for expected clean url, search URLs will actually be partially cleaned, with mobile URLs being converted to desktop ones and http being replaced with https. I hope that's still ok.

Reply:
Yes, passing `streamingpaid` as `input_relationship_type` would fix it.

And yes these small changes to search URLs are alright, as long as the error message still makes sense. (Before this kind of test existed, there were some cleanups erasing the URL or replacing it with domain name when it was not wanted.)

Reply:
And done :+1:
Now on to Melon.

Reply:
Done as well :rocket: